R/plotTuneMultiCritResult.R

```diff
@@ -68,29 +68,64 @@ plotTuneMultiCritResult = function(res, path = TRUE, col = NULL, shape = NULL, p
 #'   Visualize all evaluated points (or only the non-dominated pareto front)?
 #'   Points are colored according to their location.
 #'   Default is \code{TRUE}.
+#' @param point.info [\code{character(1)}]\cr
+#'   Show for each point which hyper parameters led to this point?
+#'   For \code{"click"}, information is displayed on mouse click.
+#'   For \code{"hover"}, information is displayed on mouse hover.
+#'   If set to \code{"none"}, no information is displayed.
+#'   Default is \code{"hover"}.
+#' @param point.trafo [\code{logical(1)}]\cr
+#'   Should the information show the transformed hyper parameters?
+#'   Default is \code{TRUE}.
 #'
 #' @template ret_ggv
 #' @family tune_multicrit
 #' @export
 #' @examples
 #' # see tuneParamsMultiCrit
-plotTuneMultiCritResultGGVIS = function(res, path = TRUE) {
+plotTuneMultiCritResultGGVIS = function(res, path = TRUE, point.info = "hover", point.trafo = TRUE) {
   requirePackages("_ggvis")
   assertClass(res, "TuneMultiCritResult")
   assertFlag(path)
+  assertChoice(point.info, choices = c("click", "hover", "none"))
+  assertFlag(point.trafo)
+
   plt.data = as.data.frame(res$opt.path)
   plt.data$location = factor(row.names(plt.data) %in% res$ind, levels = c(TRUE, FALSE),
                              labels = c("frontier", "interior"))
+  plt.data$id = seq_len(nrow(plt.data))
 
-  if (path) {
-    plt = ggvis::ggvis(plt.data, ggvis::prop("x", as.name(colnames(res$y)[1L])),
-                       ggvis::prop("y", as.name(colnames(res$y)[2L])))
-    plt = ggvis::layer_points(plt, ggvis::prop("fill", as.name("location")))
-  } else {
+  if (point.trafo) {
+    for (param in res$opt.path$par.set$pars) {
+      plt.data[, param$id] = trafoValue(param, plt.data[, param$id])
+    }
+  }
+
+  if (!path) {
     plt.data = plt.data[plt.data$location == "frontier", , drop = FALSE]
-    plt = ggvis::ggvis(plt.data, ggvis::prop("x", as.name(colnames(res$y)[1L])),
-                       ggvis::prop("y", as.name(colnames(res$y)[2L])))
-    plt = ggvis::layer_points(plt)
   }
+
+  info = function(x) {
+    if (is.null(x)) {
+      return(NULL)
+    }
+    n = length(res$x[[1]])
+    row = plt.data[plt.data$id == x$id, ][1:n]
+    text = paste0(names(row), ": ", format(row, zero.print = TRUE), collapse = "<br />")
+    return(text)
+  }
+
+  plt = ggvis::ggvis(plt.data, ggvis::prop("x", as.name(colnames(res$y)[1L])),
+    ggvis::prop("y", as.name(colnames(res$y)[2L])),
+    ggvis::prop("key", ~id, scale = FALSE))
+  plt = ggvis::layer_points(plt, ggvis::prop("fill", as.name("location")))
+
+  if (point.info == "hover") {
+    plt = ggvis::add_tooltip(plt, info, "hover")
+  }
+  else if (point.info == "click") {
+    plt = ggvis::add_tooltip(plt, info, "click")
+  }
```

Comment:
This block could be just `if (point.info != "none") { plt = ggvis::add_tooltip(plt, info, point.info) }`